database/src/unsafe_sendable.rs

```diff
@@ -0,0 +1,91 @@
+//! Wrapper type for partially-`unsafe` usage of `T: !Send`.
+
+//---------------------------------------------------------------------------------------------------- Import
+use std::{
+    borrow::Borrow,
+    ops::{Deref, DerefMut},
+};
+
+use bytemuck::TransparentWrapper;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::storable::StorableVec;
+
+//---------------------------------------------------------------------------------------------------- Aliases
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, TransparentWrapper)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[allow(clippy::unsafe_derive_deserialize)] // see safety message on the Send impl
+#[repr(transparent)]
+/// A wrapper type that `unsafe`ly implements `Send` for any `T`.
+///
+/// This is a marker/wrapper type that allows wrapping
+/// any type `T` such that it implements `Send`.
+///
+/// This is to be used when `T` is `Send`, but only in certain
+/// situations not provable to the compiler, or is otherwise a
+/// a pain to prove and/or less efficient.
+///
+/// It is up to the users of this type to ensure their
+/// usage of `UnsafeSendable` are actually safe.
+///
+/// Notably, `heed`'s table type uses this inside `service`.
+pub(crate) struct UnsafeSendable<T>(T);
+
+#[allow(clippy::non_send_fields_in_send_ty)]
+// SAFETY: Users ensure that their usage of this type is safe.
+unsafe impl<T> Send for UnsafeSendable<T> {}
+
+impl<T> UnsafeSendable<T> {
+    /// Create a new [`UnsafeSendable`].
+    ///
+    /// # Safety
+    /// By constructing this type, you must ensure the usage
+    /// of the resulting `Self` is follows all the [`Send`] rules.
+    pub(crate) const unsafe fn new(t: T) -> Self {
+        Self(t)
+    }
+
+    /// Extract the inner `T`.
+    pub(crate) fn into_inner(self) -> T {
+        self.0
+    }
+}
```

Comment:
`UnsafeSendable` impl.

The inner `T` is private so the caller must use `unsafe fn new()` and document their usage.